### ISSUE TITLE
akka.actor.AbstractProps#create should mark with type bound.

### ISSUE THREAD

**Issue by hepin1989** [akka/akka#28812]
![image](https://user-images.githubusercontent.com/501740/77607542-a935cc80-6f55-11ea-979d-fcd1ec1ed460.png)
You see, no error.
The current API:
```scala
  /**
   * Java API: create a Props given a class and its constructor arguments.
   */
  @varargs
  def create(clazz: Class[_], args: AnyRef*): Props = new Props(deploy = Props.defaultDeploy, clazz = clazz, args = args.toList)
```

**Comment by hepin1989** [comment]
Maybe we should introduce a new method `createIndirectly` and add the `[T<:Actor]` in `2.7` or `3.0`?
```scala
  /**
   * Java API: create a Props given a class and its constructor arguments.
   */
  @varargs
  def create[T<:Actor](clazz: Class[T], args: AnyRef*): Props = new Props(deploy = Props.defaultDeploy, clazz = clazz, args = args.toList)

  /**
   * Java API: create a Props given a class and its constructor arguments.
   */
  @varargs
  def createIndirectly[T<:IndirectActorProducer](clazz: Class[T], args: AnyRef*): Props = new Props(deploy = Props.defaultDeploy, clazz = clazz, args = args.toList)
```
This will help typesafe.

**Comment by johanandren** [comment]
I think it is anyway much better to go with the `create[T <: Actor](actorClass: Class[T], creator: Creator[T]):` signature. 

I don't think we should expand on those parts of the APIs. Could be good to update docs to recommend towards the non-reflection one though.

**Comment by hepin1989** [comment]
@johanandren the reflection one and the non-reflection one overlapped when the first parameter is not a subClass of Actor

**Comment by johanandren** [comment]
You mean it is not possible to use at all because it becomes an ambiguous method call?!

**Comment by hepin1989** [comment]
Yes,  @johanandren The first parameter could passed in Any class.
![image](https://user-images.githubusercontent.com/501740/78028284-58790600-7391-11ea-8ee3-5d82a919fb4a.png)
even Integer.class
![image](https://user-images.githubusercontent.com/501740/78028333-6f1f5d00-7391-11ea-8abc-443737f892ca.png)

**Comment by johanandren** [comment]
Ah, I see, it triggers the wrong overload. A mistake that will blow up in your face quite quickly though, if you try to start an arbitrary class as an actor?

**Comment by hepin1989** [comment]
Yeah, the validation just added, so it will throw an exception once you create the props.
But another problem is the API is not typesafe anymore.

The current API is been used by `reflection` for the `Dependency Injection`, and no union type support in Scala 2 and the binary compatibility, so what can we do?

**Comment by johanandren** [comment]
I agree the ergonomics are bad, but I can't see a solution that isn't very invasive (deprecate, create new methods) and I don't think it is enough of a problem to do something that invasive.

**Comment by hepin1989** [comment]
So do I, since the BC matters
